src/ddmd/mars.d

```diff
@@ -245,6 +245,52 @@ extern (C++) void genCmain(Scope* sc)
     rootHasMain = sc._module;
 }
 
+/**
+ * Copies data from a FILE* into a newly-allocated buffer.
+ *
+ * The buffer is always null-terminated.
+ */
+ubyte[] copyFile(FILE* fp)
+{
+    enum bufIncrement = 4096;
+    size_t pos = 0;
+    size_t sz = bufIncrement;
+
+    ubyte* buf = cast(ubyte*).malloc(sz + 1); // +1 for sentinel
```

Comment:
`malloc` is only meant for buffers which are freed later.
`use xmalloc` amd `xrealloc instead`
even better would be in walking the length of the input and the using allocmemory.